core/contacts.go

```diff
@@ -11,64 +15,56 @@ import (
 type ContactInfo struct {
 	Id        string      `json:"id"`
 	Address   string      `json:"address"`
-	Username  string      `json:"username"`
-	Inboxes   []repo.Cafe `json:"inboxes"`
-	Added     time.Time   `json:"added"`
-	ThreadIds []string    `json:"thread_ids"`
+	Username  string      `json:"username,omitempty"`
+	Avatar    string      `json:"avatar,omitempty"`
+	Inboxes   []repo.Cafe `json:"inboxes,omitempty"`
+	Created   time.Time   `json:"created"`
+	Updated   time.Time   `json:"updated"`
+	ThreadIds []string    `json:"thread_ids,omitempty"`
 }
 
 // AddContact adds a contact for the first time
 // Note: Existing contacts will not be overwritten
-func (t *Textile) AddContact(id string, address string, username string) error {
-	return t.datastore.Contacts().Add(&repo.Contact{
-		Id:       id,
-		Address:  address,
-		Username: username,
-		Added:    time.Now(),
-	})
+func (t *Textile) AddContact(contact *repo.Contact) error {
+	return t.datastore.Contacts().Add(contact)
 }
 
 // Contact looks up a contact by peer id
-func (t *Textile) Contact(id string) *ContactInfo {
-	model := t.datastore.Contacts().Get(id)
-	if model == nil {
-		return nil
-	}
+func (t *Textile) Contact(id string) (*ContactInfo, error) {
+	self := t.node.Identity.Pretty()
+	if id == self {
```

Comment:
This is a temp solution to get "my own contact card" in order to be able to hand it off to the peer exchange migration step. We just build up a "virtual" object from the persisted profile info. You'll want to do this _after setting username, avatar, and cafes_